api/src/models/FedoraData.ts

```diff
@@ -1,9 +1,17 @@
+export interface FedoraDatastream {
+    stream: string;
+    mimetype?: {
```

Comment:
I'm not convinced that FedoraData is the right place to put the allowedType/allowedSubtypes settings. FedoraData is for retrieving the information that is already stored in Fedora. Adding validation rules to that feels to me like it belongs in higher-level code, rather than incorporated into the data retrieval process.

I particularly feel this way because the set of datastreams ALREADY stored in Fedora is not the same as the set of datastreams ALLOWED to be stored in Fedora based on the model-to-datastream rules we have established in Geoff's spreadsheet. I think perhaps that is the piece that still needs to be implemented here: code that takes a list of models and returns a list of allowed datastreams.

In other words, rather than retrieving a list of datastreams from Fedora and then merging in mime type configurations, maybe we need a new class which takes as input configuration, a list of datastreams from Fedora, and a list of models from Fedora. Given this, it creates a comprehensive list of legal datastreams and mimetypes, and it can also flag which datastreams already have content. We can do that extra work when we need to (as part of the editing interface) but then we don't need to do extra mapping/config-reading in contexts where the constraints don't matter (e.g. when we are indexing the existing data in the system but have no intention of writing or validating anything).

Reply:
Made change.